### DISCUSSION TITLE
Milestone 2.0.11

### DISCUSSION THREAD

**Discussion by breiler** [winder/Universal-G-Code-Sender#1755] [Ideas]
What should we focus on for the next release?

There are plenty of things to work on in UGS - improving stability, improving the support for new firmwares, make things more user friendly and so on. The issue list is long so what should we focus on to be of the most use for the community?

If you have any ideas, please share your thoughts!

**Comment by chrisla23** [comment]
I would like to see a focus on visualizer usability. Here are two views of the same gcode one with UGS, and another with a different CNC control software.  This is a simple profile single pass of a 6" circle.

<img width="186" alt="Screen Shot 2021-12-27 at 10 05 23 AM" src="https://user-images.githubusercontent.com/45089130/147497030-3bee024b-222a-43db-a731-4e99cd933bea.png">

<img width="292" alt="Screen Shot 2021-12-27 at 10 05 50 AM" src="https://user-images.githubusercontent.com/45089130/147497032-23152ed0-2914-48b1-86f9-5dd18f56e3d8.png">
 
In the other package I have a simple live circle (representative of the bit width) that I can precieceisly visualize the tool/matirial in relation to where the cut will be. Getting to that view was as simple as saying zoom tool.  When I want to zoom back out, I can just say zoom extents. These are also hotkeys.

In UGS I have to awkwardly zoom and pan. This is especially clunky on my MacBook without a mouse connected. 

While the fancy spinning 3D view is neat, a simple 2D view would be of more utility.  I would also envision that view having a simple option to rotate my view (ie "rotate left 90") and keep that as a default preference as I open new files. 

I do appreciate the "Jog to Here" and snapping functionality in the current tool. I feel it would be enhanced if I knew where "here" was more exactly. 

This request was submitted entirely by my eyes, which are not getting better with age. :) 

Thanks!

-Chris

**Comment by marshr32** [comment]
Personally I'm using grbl and that's fine for what I do, so firmware isn't a big one for me. 

Also stability seems great. Nothing crashes or locks up for me.
The only major bug I see is that jogging continues after long moves with the gamepad as I've raised in issues. I've learned to expect this and work around it now though.

I think user friendliness would be great to focus on.
As Chris mentioned, the visualizer could use a little love. 
It's great being able to see things in 3d but when in use I always just put it to top view. Would be good to have UGS remember the last view when restarting.
I've noticed that if I load a file and set top view then do a jog with the gamepad, most of the visualized lines disappear leaving only the bottom layer. Or maybe they change to background colour, because from the top view the bottom layer becomes obscured and I have to rotate the model to a 3d view to see it (or reload the file)
The "set work offset to here" functionality is brilliant! I use it a lot when cutting from odd shaped pieces, and setting a point accurately is the reason I use top view. But the lines disappearing/changing does make it more difficult. I could take a video/pics of what I see if you'd like.
"Jog to here" and snapping are also brilliant.
Love the outline tool too.

UGS has overall by far the best visualizer functionality of the gcode senders that I've tried.

Could the visualizer lines that have been cut be made to disappear instead of change colour? As above, in top view the old lines tend to obscure the ones yet to be cut below. Not really a major problem as they're all on top of each other so I know where it's going to go, but a nice to have.
Thinking now, maybe I'll change the colors so cut and uncut are the same.

My other suggestion is with saving settings. When I save settings to zip it can't save everything eg my gamepad button bindings and window sizes and positions, so when I want to try a new version I need to redo them manually.

Would be nice to be able to set the DRO readout font size manually. I find it a bit small and 2.0.9 doesn't seem to scale it like the older versions.

From my view that's it. 
Are you able to post the list you mentioned? It might spark some more thoughts.

**Reply by marshr32** [reply]
Apologies, the DRO font does scale. I just hadn't played enough

**Reply by breiler** [reply]
Thanks for the input. I need to distill them into a issues to work with.

The other issues I was referring to was the big unordered [issue list](https://github.com/winder/Universal-G-Code-Sender/issues). For instance, one of the biggest issue category is regarding probing and auto leveling. Another category is to support the newer and more modern versions of GRBL (grblHAL and GrblESP32/FluidNC) to better support up to 6 axises.

**Reply by marshr32** [reply]
No problem, it's the least that we as users can do since you guys have put in so much amazing work to make UGS happen! It's really appreciated, you guys rock! 😁

I've just been playing further and found that I can go into the visualizer colors settings and make the colour of completed lines transparent using the HSL tab. Will test later today.

On the topic of saving things, it's really the UGS tab in options that I'd like to save.
There must be a file which records visualizer colors and gamepad button assignments, so they are applied at restart...... Could you tell me which file it is? 

EDIT: Just adding to this after some experimentation.
I upgraded to 2.0.10 and applied my saved settings from 2.0.9, which refreshed the screen and applied my visualiser colours but not my gamepad key bindings. However after a restart of UGS, the bindings appeared :) 
So the problem I mentioned above with the UGS tab settings is now non existent.

Re the visualiser colours: Having completed paths 100% transparent seems to work ok from a 3d view, but still obscures lower paths on top view. Will keep playing and maybe open a new discussion to keep this one on topic.

**Reply by breiler** [reply]
Regarding the settings. The way you describe how you need to handle save/load settings is an obvious usability issue that needs to be resolved. This is a built in feature to the netbeans platform that we are using and I will need to investigate how the settings should be stored/loaded between versions (I have made a note of it in #1781).

The configuration files should be stored under `C:/users/[your username>]/.ugsplatform`
